packages/headless/src/controllers/facet-manager/headless-facet-manager.ts

```diff
@@ -1,18 +1,74 @@
 import {SearchEngine} from '../../app/search-engine/search-engine';
+import {setDesiredCount} from '../../features/facets/automatic-facets/automatic-facets-actions';
+import {loadAutomaticFacetsActions} from '../../features/facets/automatic-facets/automatic-facets-actions-loader';
+import {FacetResponse} from '../../features/facets/facet-set/interfaces/response';
+import {loadReducerError} from '../../utils/errors';
 import {
   buildCoreFacetManager,
-  FacetManager,
-  FacetManagerState,
+  FacetManager as CoreFacetManager,
+  FacetManagerState as CoreFacetManagerState,
   FacetManagerPayload,
 } from '../core/facet-manager/headless-core-facet-manager';
 
-export type {FacetManagerState, FacetManagerPayload, FacetManager};
+export type {FacetManagerPayload, CoreFacetManager, CoreFacetManagerState};
+
+export interface FacetManagerState extends CoreFacetManagerState {
+  /**
+   * The list of automatic facet responses.
+   */
+  automaticFacets: FacetResponse[] | undefined;
+}
+
+export interface FacetManager extends CoreFacetManager {
+  /**
+   * The state of the FacetManager.
+   */
+  state: FacetManagerState;
+}
+
+export interface FacetManagerProps {
+  /**
+   * The desired count of automatic facets.
```

Comment:
Here and throughout, might I suggest `target`/`targetCount` rather than `desired`/`desiredCount`?
This could just be personal preference, but `desired` always feels a bit weird to me in a technical context 🤓 

I would also suggest `number`/`targetNumber` rather than `count`/`targetCount`. I think that `number` is more common in technical contexts because it's more formal.

Reply:
`desiredCount` matches the API parameter though. I'd prefer if we had the same verbiage for both

Reply:
Oh I see, looks good then 👍